### PR TITLE
Fixed a resource leak in MapPreview

### PR DESCRIPTION
**Pull request by Mailaender**
This was detected by Coverity.

### CODE REVIEW

File: OpenRA.Game/Map/MapPreview.cs [hunk 1]
```diff
@@ -253,6 +253,7 @@ public void Install()
 				{
 					Console.WriteLine(e.Message);
 					Status = MapStatus.DownloadError;
+					download.Cancel();
```
Comment (penev92) [inline OpenRA.Game/Map/MapPreview.cs]: An exception can potentially occur before line 250, which means that `download` can be `null` here. Please add a null check for extra safety.

**Comment by RoosterDragon** [comment]
The Coverity issue is reporting [```res``` is being leaked](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Map/MapPreview.cs#L217). It just needs a ```using``` on it as far as I can tell.

I don't think the current change of cancelling the download makes sense.

**Comment by Mailaender** [comment]
Updated to fix the real issue.

**Comment by RoosterDragon** [comment]
:+1:

**Comment by penev92** [comment]
:+1:

**Pull request merged**